Client/TabManagement/Legacy/LegacyTabManager.swift

```diff
@@ -898,7 +898,7 @@ extension LegacyTabManager: WKNavigationDelegate {
                 return
             }
 
-            if let title = webView.title {
+            if let title = webView.title, selectedTab?.webView == webView {
                 selectedTab?.lastTitle = title
             }
```

Comment:
I'm not really sure if this is really needed 🤔. The issue is that when opening a new tab is updating the title from the original tab 

cc: @OrlaM in case you have more context